### ISSUE TITLE
Chroma Key Color Issue

### ISSUE THREAD

**Issue by cha1r** [lay295/TwitchDownloader#1021]
### Checklist

- [X] I have checked the [issue page](https://github.com/lay295/TwitchDownloader/issues?q=is%3Aissue) for duplicates

### Write stuff here

Hi, I have been trying to render chats, but I'm having no luck finding a compatible, Chroma key color. Green obviously filters out most pepe emotes and user names, and nearly every other color does the same.

Is there an option that auto removes the background and removes the need for a Chroma key? If not, what color should I be using?

**Comment by ScrubN** [comment]
> Is there an option that auto removes the background and removes the need
for a Chroma key? If not, what color should I be using?

Actually there is. In the Windows GUI, it is called "generate mask" and it
will produce a black and white video file that can be used as a source for
a mask effect in your video editor.

**Comment by ScrubN** [comment]
The mask also depends on the background color having an alpha channel, so
don't forget to set that.

Don't worry if your output codec doesn't support alpha channels, FFmpeg
will take care of it.

**Comment by cha1r** [comment]
Thank you for the response! I tried what you said and I was successfully able to export a black and white mask alongside an exported chat render, however I'm unsure how I would go about using said mask for the chroma key, and the chat render that came with it still has a solid background. If its not too much trouble, could you describe a step by step process? I fear I might be misunderstanding your instructions.

**Comment by ScrubN** [comment]
Most video editors have a masking effect/filter which goes on the source footage and takes in a black & white mask that is used to selectively apply transparency to said source footage.

If you are using Sony Vegas, masking is done via compositing. This video explains it well https://youtu.be/8fNTUEEFhdY, but you will need to add a mask generator effect to the mask source set to luminance because it does not contain an alpha channel.

**Comment by cha1r** [comment]
Thank you again for your response! I'm still struggling a bit here. 
 
Here is what I've gathered, let me know if I'm missing something

1. Generate mask - used for source for mask 
2. background color must have alpha channel 
3. composite of some kind for mask - need a mask generator effect to the mask source

Heres what I've been doing. Currently I am exporting my chats and leaving only the A channel at around half transparency, with all other colors set to 0, once exported I get two video files, one regular text chat with a black background, full color emotes ect, and one grey and white mirror file, which I assume is the mask that was generated. From there, I want to bring that into my editor (premier) and use some form of mask generator to eliminate the solid black of my original chat (the one with a solid black background which I seek to make transparent) using the generated mask. Note I also tried this with a mask export that is solid black, and white text + emotes, and I'm not having luck with it either. 

I guess I am failing to see how the generated mask comes into play here. I get that its needed for the selective transparency, I just fail to see how the black and white base achieves this as my primary chat remains untouched. 

I'm sorry to take up your time with this, and thank you for all of your help
